python/dgl/backend/pytorch/tensor.py

```diff
@@ -135,15 +135,15 @@ def zeros_like(input):
 def ones(shape, dtype, ctx):
     return th.ones(shape, dtype=dtype, device=ctx)
 
-if TH_VERSION.version[0] == 0:
-    # TODO(minjie): note this does not support autograd on the `x` tensor.
-    #               should adopt a workaround using custom op.
-    def spmm(x, y):
-        return th.spmm(x, y)
-else:
-    # torch v1.0+
-    def spmm(x, y):
-        return th.sparse.mm(x, y)
+def spmm(x, y):
+    dst, src = x._indices()
+    # scatter index
+    index = dst.view(-1, 1).expand(-1, y.shape[1])
+    # zero tensor to be scatter_add to
+    out = y.new_full((x.shape[0], y.shape[1]), 0)
+    # look up src features and multiply by edge features
+    feature = th.index_select(y, 0, src) * x._values().unsqueeze(-1)
```

Comment:
Add a note above about the `index_select` and `[]` operator pitfall.

Reply:
Done